pkg/pillar/cmd/logmanager/logmanager.go

```diff
@@ -192,29 +195,35 @@ func Run(ps *pubsub.PubSub) {
 		globalConfig: &types.GlobalConfigDefaults,
 	}
 	// Look for global config such as log levels
-	subGlobalConfig, err := pubsublegacy.Subscribe("", types.GlobalConfig{},
-		false, &logmanagerCtx, &pubsub.SubscriptionOptions{
-			CreateHandler: handleGlobalConfigModify,
-			ModifyHandler: handleGlobalConfigModify,
-			DeleteHandler: handleGlobalConfigDelete,
-			WarningTime:   warningTime,
-			ErrorTime:     errorTime,
-		})
+	subGlobalConfig, err := ps.NewSubscription(pubsub.SubscriptionOptions{
+		AgentName:     "",
+		TopicImpl:     types.GlobalConfig{},
+		Activate:      false,
+		Ctx:           &logmanagerCtx,
+		CreateHandler: handleGlobalConfigModify,
+		ModifyHandler: handleGlobalConfigModify,
+		DeleteHandler: handleGlobalConfigDelete,
+		WarningTime:   warningTime,
+		ErrorTime:     errorTime,
+	})
 	if err != nil {
 		log.Fatal(err)
 	}
 	logmanagerCtx.subGlobalConfig = subGlobalConfig
 	subGlobalConfig.Activate()
 
 	// Get DomainStatus from domainmgr
-	subDomainStatus, err := pubsublegacy.Subscribe("domainmgr",
-		types.DomainStatus{}, false, &logmanagerCtx, &pubsub.SubscriptionOptions{
-			CreateHandler: handleDomainStatusModify,
-			ModifyHandler: handleDomainStatusModify,
-			DeleteHandler: handleDomainStatusDelete,
-			WarningTime:   warningTime,
-			ErrorTime:     errorTime,
-		})
+	subDomainStatus, err := ps.NewSubscription(pubsub.SubscriptionOptions{
+		AgentName:     "domainmgr",
+		TopicImpl:     types.DomainStatus{},
+		Activate:      false,
+		Ctx:           logmanagerCtx,
```

Comment:
Missing & results in panic with 
2020-03-12T14:28:18.490975+00:00 linuxkit-525400123456 pillar.err 2020-03-12T14:28:18Z,pillar.err;panic: interface conversion: interface {} is logmanager.DNSContext, not *logmanager.DNSContext
2020-03-12T14:28:18.683559+00:00 linuxkit-525400123456 pillar.err 2020-03-12T14:28:18Z,pillar.err;github.com/lf-edge/eve/pkg/pillar/cmd/logmanager.handleDNSModify(0x1b33b80, 0xc00013e9f0, 0xc00044901a, 0x6, 0x1b02800, 0xc00000e020)
2020-03-12T14:28:18.735955+00:00 linuxkit-525400123456 pillar.err 2020-03-12T14:28:18Z,pillar.err;	/pillar/cmd/logmanager/logmanager.go:418 +0x5a5
2020-03-12T14:28:19.028682+00:00 linuxkit-525400123456 pillar.err 2020-03-12T14:28:18Z,pillar.err;github.com/lf-edge/eve/pkg/pillar/cmd/logmanager.Run(0xc000438000)
2020-03-12T14:28:19.075908+00:00 linuxkit-525400123456 pillar.err 2020-03-12T14:28:18Z,pillar.err;	/pillar/cmd/logmanager/logmanager.go:273 +0x144d